src/fr-auth/callbacks/device-profile-callback.ts

```diff
@@ -0,0 +1,45 @@
+import FRCallback from '.';
+import { Callback } from '../../auth/interfaces';
+import { DeviceProfileData } from '../../fr-device/interfaces';
+
+/**
+ * Represents a callback used to collect device profile data.
+ */
+class DeviceProfileCallback extends FRCallback {
+  /**
+   * @param payload The raw payload returned by OpenAM
+   */
+  constructor(public payload: Callback) {
+    super(payload);
+  }
+
+  /**
+   * Gets the callback's data.
+   */
+  public getMessage(): string {
+    return this.getOutputByName<string>('message', '');
+  }
+
+  /**
+   * Does callback require metadata?
+   */
+  public isMetadataRequired(): boolean {
```

Comment:
The `get/set` prefixing is really a Java thing.  IMO, the `is` prefix is superior for signaling intent that this returns a boolean. `getMetadataRequired` sounds like it would determine required metadata and return it.

Reply:
Yeah, the sole purpose of this is the resulting `boolean`, which is separate from the actual "metadata" itself. So, I feel the `is` is more indicative of that intent.